src/Bridge/Faker/Provider/EnumProvider.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the "elao/enum" package.
+ *
+ * Copyright (C) 2017 Elao
+ *
+ * @author Elao <contact@elao.com>
+ */
+
+namespace Elao\Enum\Bridge\Faker\Provider;
+
+use Elao\Enum\EnumInterface;
+
+class EnumProvider
+{
+    /**
+     * Enum mapping as an array with :
+     *
+     * - alias for Enum class as key
+     * - Enum FQCN as value
+     *
+     * Example :
+     *
+     * ```
+     * [
+     *   'Civility' => Civility::class,
+     *   'Gender'   => Gender::class,
+     * ]
+     * ```
+     *
+     * @var array $enumMapping
+     */
+    private $enumMapping = [];
+
+    public function __construct(array $enumMapping)
+    {
+        foreach ($enumMapping as $enumAlias => $enumClass) {
+            $this->ensureEnumClass($enumClass);
+            $this->enumMapping[$enumAlias] = $enumClass;
+        }
+    }
+
+    /**
+     * @param string $enumValueShortcut As <ENUM_CLASS_ALIAS::ENUM_VALUE_CONSTANT>
+     *                                  Examples: 'Gender::MALE', 'Gender::FEMALE', 'Permissions::READ|WRITE', etc.
+     *
+     * @throws \InvalidArgumentException When the alias part of $enumValueShortcut is not a valid alias
+     *
+     * @return EnumInterface
+     */
+    public function enum(string $enumValueShortcut): EnumInterface
+    {
+        list($classAlias, $constants) = explode('::', $enumValueShortcut);
+
+        if (!array_key_exists($classAlias, $this->enumMapping)) {
+            throw new \InvalidArgumentException("$classAlias is not a valid alias");
+        }
+
+        /** @var EnumInterface $class */
+        $class = $this->enumMapping[$classAlias];
+
+        $constants = explode('|', $constants);
```

Comment:
Perhaps we should check if the enum class is an instance of `FlaggedEnum` to throw a nice exception otherwise